### PR TITLE
Generic library documentation update

### PR DESCRIPTION
**Pull request by pekkaklarck**
Includes explaining the new `strategy:value` syntax (#908).

Still plan to update the documentation of the `__init__`, but going through individual keywords is too much work for 3.0.

### CODE REVIEW

File: src/SeleniumLibrary/__init__.py [hunk 1]
```diff
@@ -42,133 +42,190 @@
...
+    | `Click Link`    | example | # Match also based on link text and ``href``.   |
+    | `Click Button`  | example | # Match based on ``id``, ``name`` or ``value``. |
+
+    If the locator accidentally starts with some of the explicit locator
+    prefixes or with the implicit XPath prefixes, it is possible to use
+    the explicit ``default`` prefix to enable the default strategy.
+
+    Examples:
+
+    | `Click Element` | name:foo         | # Find element with class ``foo``.               |
```
Comment (aaltat) [inline src/SeleniumLibrary/__init__.py]: I thin comment should say: 
```
Find element with name attribute ``foo``
```

File: src/SeleniumLibrary/__init__.py [hunk 2]
```diff
@@ -42,133 +42,190 @@
...
+    | `Click Element` | default:name:foo | # Use default strategy with value ``class:foo``. |
+    | `Click Element` | //foo            | # Find element using XPath ``//foo``.            |
+    | `Click Element` | //foo            | # Use default strategy with value ``//foo``.     |
```
Comment (aaltat) [inline src/SeleniumLibrary/__init__.py]: The locator is same in the above xpath example but the comment says something else, should the locator be just `foo` without the double forward slashes.

File: src/SeleniumLibrary/__init__.py [hunk 3]
```diff
@@ -42,133 +42,190 @@
...
 
     = Timeouts =
 
-    There are several `Wait ...` keywords that take timeout as an
+    There are several ``Wait ...`` keywords that take timeout as an
     argument. All of these timeout arguments are optional. The timeout
-    used by all of them can be set globally using the
-    `Set Selenium Timeout` keyword. The same timeout also applies to
-    `Execute Async Javascript`.
+    used by all of them can be set globally using the `Set Selenium Timeout`
```
Comment (aaltat) [inline src/SeleniumLibrary/__init__.py]: Should we also say that timeout can be set in the library import?

**Review by aaltat — CHANGES_REQUESTED** [review]

File: src/SeleniumLibrary/__init__.py (hunk 1 above)
Comment (pekkaklarck) [reply src/SeleniumLibrary/__init__.py]: Yeah, I changed the example but apparently forget the comment. Will fix.

File: src/SeleniumLibrary/__init__.py (hunk 2 above)
Comment (pekkaklarck) [reply src/SeleniumLibrary/__init__.py]: It should be `default://foo`. Will fix.

File: src/SeleniumLibrary/__init__.py (hunk 3 above)
Comment (pekkaklarck) [reply src/SeleniumLibrary/__init__.py]: Good idea. Will add.

**Review by aaltat — APPROVED** [review]

**Pull request merged**